src/main/java/com/bean00/server/RequestProcessor.java

```diff
@@ -17,27 +17,64 @@ public RequestProcessor(DataStore dataStore) {
     }
 
     public Response processRequest(Request request) throws IOException {
+        String requestMethod = request.getRequestMethod();
+        Response response;
+
+        if (requestMethod.equals(Method.PUT)) {
+            response = putHandler(request);
+        } else if (requestMethod.equals(Method.GET) || requestMethod.equals(Method.HEAD)) {
+            response = getHeadHandler(request);
+        } else {
+            response = methodNotAllowedResponse("GET, HEAD, PUT");
+        }
+
+        return response;
+    }
+
+    private Response putHandler(Request request) throws IOException {
+        String requestURL = request.getRequestURL();
+        byte[] requestBody = request.getBodyAsBytes();
+
+        if (dataStore.isDirectory(requestURL)) {
+            return methodNotAllowedResponse("GET, HEAD");
+        }
+
+        int statusCode = dataStore.resourceExists(requestURL) ? Status.OK : Status.CREATED;
+
+        dataStore.put(requestURL, requestBody);
+
+        return new Response(statusCode);
+    }
+
+    private Response methodNotAllowedResponse(String allowedMethods) {
```

Comment:
similar thing here as my previous comment. Try to use a verb. Also, I would try and keep the naming similar to the other methods you use in the same if/else block

Reply:
Renamed